sensors/jira_sensor.py

```diff
@@ -103,12 +103,12 @@ def _detect_new_issues(self):
     def _dispatch_issues_trigger(self, issue):
         trigger = self._trigger_ref
         payload = {}
+        payload['id'] = issue.id
         payload['issue_name'] = issue.key
         payload['issue_url'] = issue.self
-        payload['issue_browse_url'] = self._jira_url + '/browse/' + issue.key
         payload['project'] = self._project
-        payload['created'] = issue.raw['fields']['created']
-        payload['assignee'] = issue.raw['fields']['assignee']
-        payload['fix_versions'] = issue.raw['fields']['fixVersions']
-        payload['issue_type'] = issue.raw['fields']['issuetype']['name']
+        payload['expand'] = issue.raw['expand']
```

Comment:
The code you have linked to is setting the `fields` and `expand` keys in the _request_ to Jira.

That implies nothing as to whether they are included in the response from Jira.

However, according to the [Jira REST API examples](https://developer.atlassian.com/server/jira/platform/jira-rest-api-examples/) it does appear that at least `fields` is not returned in all responses from Jira (although it is returned in some of them, see the second code example in [this section](https://developer.atlassian.com/server/jira/platform/jira-rest-api-examples/#calling-the-rest-api-from-a-script-graphing-image-links)).

So you are correct: it would be better to use `issue.raw.get('fields', None)`, or perhaps `issue.raw.get('fields', {})` or `issue.raw.get('fields', [])` - whichever one is more appropriate to indicate an empty result.

For `expand`, it also does not seem to appear in all results, so it too should be accessed with `.get()`. The default value for it should be a simple empty string (`""`) since it will always be a string composed of comma-separated strings.